### PR TITLE
FirstOf: added extra-ctor

### PR DESCRIPTION
**Pull request by iakunin**
This pull-request has no corresponding issue.

* Added extra-ctor with default `condition` (always `true`) to FirstOf

### CODE REVIEW

**Comment by iakunin** [comment]
@paulodamaso could you have a look, please?

**Comment by 0crat** [comment]
This pull request [#1365](https://github.com/yegor256/cactoos/issues/1365) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/C63314D6Z/4aa754c0-d9bf-4fb8-a357-66e0956aeb56); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

**Comment by victornoel** [comment]
@iakunin can you add a link to the original issue being solved here? (or maybe there isn't one?)

File: src/main/java/org/cactoos/scalar/FirstOf.java [hunk 1]
```diff
@@ -53,6 +53,19 @@
      */
     private final Scalar<T> fallback;
 
+    /**
+     * Constructor with default condition (always `true`).
+     * @param src Source iterable
+     * @param fbck Fallback used if no value matches
+     */
+    public FirstOf(final Iterable<T> src, final Scalar<T> fbck) {
```
Comment (victornoel) [inline src/main/java/org/cactoos/scalar/FirstOf.java:61]: @iakunin let's also add a constructor that takes a `T` to ease use, WDYT?

**Comment by iakunin** [comment]
> @iakunin can you add a link to the original issue being solved here? (or maybe there isn't one?)

@victornoel there isn't one =)

File: src/main/java/org/cactoos/scalar/FirstOf.java (hunk 1 above)
Comment (iakunin) [reply src/main/java/org/cactoos/scalar/FirstOf.java:61]: @victornoel do you mean `T` as a fallback, right?

Comment (victornoel) [reply src/main/java/org/cactoos/scalar/FirstOf.java:61]: @iakunin yes exactly, sorry for the confusion :)

Comment (iakunin) [reply src/main/java/org/cactoos/scalar/FirstOf.java:74]: @victornoel done =)

File: src/test/java/org/cactoos/scalar/FirstOfTest.java [hunk 2]
```diff
@@ -118,4 +118,28 @@ public void throwsFallbackIfNothingMatches() {
...
+    @Test
+    public void returnsFirstValueWithIntegerFallback() {
+        new Assertion<>(
+            "Returns first value with Integer fallback",
+            new FirstOf<>(
+                new IterableOfInts(2, 1, 0),
+                -1
+            ),
+            new ScalarHasValue<>(2)
+        ).affirm();
```
Comment (victornoel) [inline src/test/java/org/cactoos/scalar/FirstOfTest.java:143]: @iakunin can you just add one test for the empty iterable version?

Comment (iakunin) [reply src/test/java/org/cactoos/scalar/FirstOfTest.java:143]: @victornoel sure. Done.

**Comment by victornoel** [comment]
@iakunin thx :)

@paulodamaso looks good to me

**Comment by paulodamaso** [comment]
@iakunin Thank you for the PR

**Comment by paulodamaso** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 10 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @victornoel[/z](https://www.0crat.com/u/victornoel), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good
